CONTRIBUTING.md

```diff
@@ -1,86 +1,187 @@
+
+If you're only fixing a bug, it's fine to submit a pull request right away but we still recommend filing an issue detailing what you're fixing. This is helpful in case we don't accept that specific fix but want to keep track of the issue.
+
+### Feature requests
+
+If you would like to request a new feature or enhancement but are not yet thinking about opening a pull request, you can file an issue with the [feature template](https://github.com/pact-foundation/pact-python/issues/new?assignees=&labels=feature%2Cstatus%3A+needs+triage&template=feature.yml) for more thought out ideas. Alternatively, you can use the [Canny board](https://pact.canny.io/) for more casual feature requests and gain enough traction before proposing an RFC.
+
+### Claiming issues
+
+We have a list of [beginner-friendly issues](https://github.com/pact-foundation/pact-python/labels/good%20first%20issue) to help you get your feet wet in the Pact Python codebase and familiar with our contribution process. This is a great place to get started.
+
+Apart from the `good first issue`, it is also worth looking at the [`help wanted`](https://github.com/pact-foundation/pact-python/labels/help%20wanted) issues. If you have specific knowledge in one domain, working on these issues can make your expertise shine.
+
+If you want to work on any of these issues, just drop a message saying "I'd like to work on this", and we will assign the issue to you and update the issue's status as "claimed".
+
+Alternatively, when opening an issue, you can also click the "self service" checkbox to indicate that you'd like to work on the issue yourself, which will also make us see the issue as "claimed".
+
+## Development
+
+### Online one-click setup for contributing
+
+You can also try using the new [github.dev](https://github.dev/pact-foundation/pact-python) feature. While you are browsing any file, changing the domain name from `github.com` to `github.dev` will turn your browser into an online editor. You can start making changes and send pull requests right away.
+
+### Installation
+
+1.  Ensure you have [Python](https://www.python.org/) installed.
+
+2.  Ensure you have [Hatch](https://hatch.pypa.io/) installed. This is used to manage the development environment can be installed as follows:
+
+    ```sh
+    python -m pip install --user pipx # If you don't have pipx
+    pipx install hatch
+    ```
+
+3.  After cloning the repository, run `hatch shell` in the root of the repository. This will install all dependencies in a Python virtual environment and then ensure that the virtual environment is activated.
+
+4.  To run tests, run `hatch run test` to make sure the test suite is working. You should also make sure the example works by running `hatch run example`. For the examples, you will have to make sure that you have Docker (or a suitable alternative) installed and running.
```

Comment:
is there a way just to run a subset of the examples

Reply:
For the tests, definitely. Given the tests don't take all that long to run, I don't know whether it is going into too much detail in this document. Having said that, `hatch run test {args}` effectively aliases to `pytest <some default args> {args}`. So `hatch run test tests/some/test_file.py` would run just the tests discovered in this file.

As for the example, it actually isn't really possible to run them individually. The consumer side _has_ to run first and the provider are tested against the consumer. This is why I also numbered the tests, to help ensure the ordering is correct. I am very well aware that this is a bit idiosyncratic for a test suite, but given it isn't part of the unit testing and serves purely as an example, I thought this could be excused.

Reply:
I'm okay with that because I could just run the consumer test first indiv, and the the provider (in the case of testing out the message example) - it would just save me about 30 seconds per run, when I was just identifying an issue with the message example tests, and I love quick feedback loops